src/conversions/std/ipaddr.rs

```diff
@@ -0,0 +1,123 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use crate::exceptions::PyValueError;
+use crate::once_cell::GILOnceCell;
+use crate::types::PyType;
+use crate::{intern, FromPyObject, IntoPy, Py, PyAny, PyObject, PyResult, Python, ToPyObject};
+
+impl GILOnceCell<Py<PyType>> {
+    pub(crate) fn get_or_try_init_type_ref<'py>(
+        &'py self,
+        py: Python<'py>,
+        module_name: &'static str,
+        attr_name: &'static str,
+    ) -> PyResult<&'py PyType> {
+        self.get_or_try_init(py, || py.import(module_name)?.getattr(attr_name)?.extract())
+            .map(|ty| ty.as_ref(py))
+    }
+}
+
+static IPV4_ADDRESS: GILOnceCell<Py<PyType>> = GILOnceCell::new();
```

Comment:
I think those statics could have their scope reduced by moving them into the `to_object` bodies.